components/markdown_image.jsx

```diff
@@ -50,14 +53,34 @@ export default class MarkdownImage extends React.PureComponent {
         this.setState({loadFailed: true});
     }
 
+    handleImageLoaded = ({height, width}) => {
+        if (this.state.loadFailed) {
+            this.setState({loadFailed: false});
+        }
+        this.props.onImageLoaded({height, width});
+    }
+
+    isHeaderChangeMessage = () => {
+        return this.props.postType &&
+            this.props.postType === Constants.PostTypes.HEADER_CHANGE;
+    }
+
     render() {
         const {imageMetadata, src, alt, imageIsLink} = this.props;
         if (src === '' || this.state.loadFailed) {
+            let className = 'markdown-inline-img broken-image';
+            if (this.isHeaderChangeMessage()) {
+                className += ' broken-image--scaled-down';
+            }
+
             return (
-                <img
-                    alt={alt}
-                    src={brokenImageIcon}
-                />
+                <div className={'style--none'}>
```

Comment:
I see that you already posted inline images with failed condition but i am confused about how they are inline if we are changing the wrapper to a div? why are we changing this and adding a class `style-none`?

Reply:
Good point! MD images always had an inline container, whether from the `SizeAwareImage` or from a `markdown__paragraph-inline` class as with the case of the channel header and the system messages (the posted examples you are referring to). However, I'd missed the case that posting a chat message with a broken image will actually be block-displayed, sorry for the confusion. I will fix this.

As for the `style--none`, the first time I encountered it was in the `SizeAwareImage`, here: 

```
<div
        className='style--none file-preview__button'
        style={imageStyleChangesOnLoad}
>
```

And I'd adopted a similar approach while working on [MM-12898](https://github.com/mattermost/mattermost-webapp/pull/3639), but admittedly, I hadn't looked into this class in details. Now that you mentioned it, I realized that it is meant to be used on buttons and has no effect on `div` elements. So, I will change this in MD images, and will also remove this class from `SizeAwareImage`.